services/ui-src/src/routes/Dashboard/Dashboard.tsx

```diff
@@ -74,54 +71,47 @@ export const Dashboard = () => {
     localStorage.setItem("selectedReport", "");
   }, []);
 
-  const enterSelectedReport = async (reportId: string) => {
+  const enterSelectedReport = async (reportMetadata: ReportShape) => {
     // set active report to selected report
     const reportDetails: ReportDetails = {
-      state: activeState!,
-      reportId: reportId,
+      state: reportMetadata.state!,
+      reportId: reportMetadata.reportId,
     };
     setReport(reportDetails);
-    localStorage.setItem("selectedReport", reportId);
+    localStorage.setItem("selectedReport", reportMetadata.reportId);
 
     // fetch & set active report to selected report
     await fetchReport(reportDetails);
     const reportFirstPagePath = "/mcpar/program-information/point-of-contact";
     navigate(reportFirstPagePath);
   };
 
-  const openAddEditProgramModal = (reportId?: string) => {
-    // if reportId provided, set as selected program
-    setSelectedReportId(reportId);
-
-    const selectedReport = reportsByState?.find(
-      (o: { reportId: string }) => o.reportId === reportId
-    );
+  const openAddEditProgramModal = (reportMetadata?: ReportShape) => {
+    let formData = undefined;
     // Check and pre-fill the form if the user is editing an existing program
-    if (reportId && selectedReport) {
-      const formData = {
+    if (reportMetadata) {
+      formData = {
         fieldData: {
-          "aep-programName": selectedReport.programName,
+          "aep-programName": reportMetadata.programName,
           "aep-endDate": convertDateUtcToEt(
-            selectedReport.reportingPeriodEndDate
+            reportMetadata.reportingPeriodEndDate
           ),
           "aep-startDate": convertDateUtcToEt(
-            selectedReport.reportingPeriodStartDate
+            reportMetadata.reportingPeriodStartDate
           ),
         },
         state: activeState,
```

Comment:
Should this read from reportMetadata as well? They should always be the same as activeState, but since we're reading everything else from there

Reply:
Good catch! Double checking that didn't break nothing